tests/core/nodes/test_tools.py

```
@@ -0,0 +1,302 @@
+        bash2 = BashNode(id="bash2", session=session)
+
+        tools, executor = tools_from_nodes([bash1, bash2])
+
+        assert len(tools) == 2
+        tool_names = {t.name for t in tools}
+        assert tool_names == {"bash1", "bash2"}
+
+    @pytest.mark.asyncio
+    async def test_executor_dispatches_correctly(self, session):
+        """Executor routes to correct node."""
+        bash1 = BashNode(id="bash1", session=session, cwd="/tmp")
```

Ruff (0.14.10)

Probable insecure usage of temporary file or directory: "/tmp"

(S108)